pkg/sql/catalog/descs/collection.go

```diff
@@ -380,6 +380,15 @@ func (tc *Collection) GetTableVersion(
 		// know how to deal with, so propagate the error.
 		return nil, err
 	}
+	table, ok := desc.(*sqlbase.ImmutableTableDescriptor)
+	if !ok {
+		// TODO (lucy): Is this the right error? I think it should be possible at
```

Comment:
I think we have some errors like "ID is not a table" that we use in these cases.

Reply:
But table does not exist when you find a type is a fine error